step/src/main/java/com/google/sps/User.java

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+public class User {
+  
+  private long id;
+  private String email;
+  private String name;
+  private Collection<User> matches;
+
+  // TODO: determine what fields we want.
+  // TODO: update the constructor so that it only needs to take
+  //       in an id, and the other fields can then be looked up
+  //       in the database. this update will be made when data-
+  //       -store is integrated
+  public User(long id, String email, String name) {
+    this.id = id;
+    this.email = email;
+    this.name = name;
+    //TODO: update this to fetch from database instead of making a new hashset    
+    this.matches = new HashSet<>();
+  }
```

Comment:
With how I'm imagining a User object being created, I think it would be best to eliminate the id parameter and have the User class maintain a static id variable that is incremented and assigned with every call to the constructor. This would of course depend on the use case of the User object. What do you think?

Reply:
I think this makes sense but would be better suited for the sign-up class where users IDs are being generated and assigned. Each instance of the User class represents one individual user and so we want to use it to store their permanent unchanging ID. 

Also, in terms of generating IDs, having a number that we increment on for each new user makes sense. For security purposes we might want to look into generating random numbers for the IDs and maybe storing all the IDs to compare against and make sure that each newly generated ID hasn't been used before. We could also make use of an API that generated user-IDs, like this one: http://www.javapractices.com/topic/TopicAction.do?Id=56